infra-as-code/bicep/modules/vwanConnectivity/vwanConnectivity.bicep

```diff
@@ -48,24 +88,25 @@ param parVirtualWanHubName string = '${parCompanyPrefix}-vhub'
 - `parHubRoutingPreference` - The Virtual WAN Hub routing preference. The allowed values are `ASN`, `VpnGateway`, `ExpressRoute`.
 - `parVirtualRouterAutoScaleConfiguration` - The Virtual WAN Hub capacity. The value should be between 2 to 50.
 - `parVirtualHubRoutingIntentDestinations` - The Virtual WAN Hub routing intent destinations, leave empty if not wanting to enable routing intent. The allowed values are `Internet`, `PrivateTraffic`.
+- `parUseCustomNamingScheme` - Switch to enable/disable custom naming scheme. When enabled a custom name can be given for Azure Firewall, ExpressRoute Gateway, VPN Gateway and Virtual Hubs.
 
 ''')
-param parVirtualWanHubs array = [ {
+param parVirtualWanHubs virtualWanOptionsType = [ {
     parVpnGatewayEnabled: true
     parExpressRouteGatewayEnabled: true
     parAzFirewallEnabled: true
     parVirtualHubAddressPrefix: '10.100.0.0/23'
     parHubLocation: parLocation
-    parHubRoutingPreference: 'ExpressRoute' //allowed values are 'ASN','VpnGateway','ExpressRoute'.
-    parVirtualRouterAutoScaleConfiguration: 2 //minimum capacity should be between 2 to 50
+    parHubRoutingPreference: 'ExpressRoute'
+    parVirtualRouterAutoScaleConfiguration: 2
     parVirtualHubRoutingIntentDestinations: []
   }
 ]
 
-@sys.description('Prefix Used for VPN Gateway.')
+@sys.description('VPN Gateway Name.')
 param parVpnGatewayName string = '${parCompanyPrefix}-vpngw'
 
-@sys.description('Prefix Used for ExpressRoute Gateway.')
+@sys.description('ExpressRoute Gateway Name.')
```

Comment:
This is still prefix, so no need to change the description.

Reply:
@picccard I don't I agree. I see where you are coming from, but the name of the parameter is `parExpressRouteName` or `parVpnGatewayName` so that's why I changed the description. I said that I see where you are coming from because it is used as a prefix in the 'name' property. I don't mind changing it back to prefix though.

@oZakari What are your thoughts?

Reply:
I agree the parameter name is bad/misleading, that is why i think the description should stay as before 🤷‍♂️

Reply:
Agreed that it's a bit misleading. I was kind of going back on forth on this, but I think it's probably best to keep the description set as the prefix. I think it provides a bit more context and may avoid frustration if people see the description and expect it to be the actual name once it's provisioned. 

Although this would be a breaking change, we could create a backlog item to restructure/rename that param long term as well. Thoughts?